### DISCUSSION TITLE
v20230308 ValueError: Expected parameter logits

### DISCUSSION THREAD

**Discussion by plasmaphase** [openai/whisper#1068] [Q&A]
I've noticed in the last several versions that with longer audio files I occasionally (but consistently) get the following error:

```bash
ValueError: Expected parameter logits (Tensor of shape (1, 51865)) of distribution Categorical(logits: torch.Size([1, 51865])) to satisfy the constraint IndependentConstraint(Real(), 1), but found invalid values:
tensor([[nan, nan, nan,  ..., nan, nan, nan]], device='cuda:0')
```

Some files process to completion, but the ones that don't fail at the same moment consistently.

**Comment by jongwook** [comment]
Hi, are you using the default arguments in the command line, and would you be able to share an audio file that fails consistently?

**Reply by plasmaphase** [reply]
Yes, using the default arguments, medium model (but have tried base as well).  The files that fail are pretty huge, 1/2 a gig and beyond, so probably not, but let me see what I can do.

**Reply by jongwook** [reply]
For now, setting `--temperature_increment_on_fallback None` will probably prevent this issue by disabling nonzero temperatures, since that's when it samples from the Categorical distribution.

**Reply by plasmaphase** [reply]
Sorry for my lack of knowledge, but how would I add that parameter using python (not using cli)?

**Reply by jongwook** [reply]
In Python, you can fix the temperature at zero by: `model.transcribe(..., temperature=0)` (it's `(0.0, 0.2, 0.4, 0.6, 0.8, 1.0)` if you don't specify this)

**Reply by ai-bits** [reply]
`whisper audio.m4a --model medium --language English --temperature_increment_on_fallback None`

With --temperature.. the error is gone, but output is still trash.
Cross-posting from "ValueError Tensor on external RTX 2070" https://github.com/openai/whisper/discussions/1248
TIA for any further insights!

**Reply by NuperSu** [reply]
I also got this error (used medium model and ru lang). With `--temperature_increment_on_fallback None`, it run without errors but outputted:
```
[00:29.240 --> 00:59.240] """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""#
```

**Reply by Jacky-MYQ** [reply]
Though this error diappeared by using this way, my txt became  "!!!!!!!!!!!!!!!!!" which is not my video voice. However, i try to add "fp16=False" in this code `model.transcribe("..",fp16=False)` . I find out this mistake diappears `ValueError: Expected parameter logits (Tensor of shape (1, 51865)) of distribution Categorical(logits: torch.Size([1, 51865])) to satisfy the constraint IndependentConstraint(Real(), 1), but found invalid values:
tensor([[nan, nan, nan,  ..., nan, nan, nan]], device='cuda:0')` and i got a useful txt! So, the problem is settled!

**Reply by AriyaRasekh** [reply]
Any update on this?

**Comment by DmitrySandalov** [comment]
Try using `fp16=False` in Python and `--fp16 False` in the command line.

**Reply by junshichen** [reply]
this solution works.

**Reply by janngobble** [reply]
Yes, this fixed my issues.
